arch/arm/src/armv7-m/arm_itm_syslog.c

```diff
@@ -106,8 +107,10 @@ static const struct syslog_channel_s g_itm_channel =
  *
  ****************************************************************************/
 
-static int itm_putc(int ch)
+static int itm_putc(void *dev, int ch)
 {
+  (void)dev;
```

Comment:
```suggestion
  UNUSED(dev);
```
You may use the UNUSED macro from `nuttx/compiler.h`.

Reply:
I think this is not covered in coding standard.  
What is the preferred style in NuttX?

Also, maybe someone to add this in coding standard?

Reply:
I believe this is something that should not be covered in the NuttX Style Guide.
It is more of a code quality question, to make the code cleaner.

Reply:
AFAIK `(void) unused_variable;` is perfectly valid in C, with no portability issues.  
Or if I am wrong, please enlighten me.

So this is why I consider this a "styling" issue.

Nevertheless, I will update all occurrences of this to use the `UNUSED()` macro. :)

Reply:
It is a matter of consistency with the rest of the codebase. Also, I think the macro is more verbose in its purpose, that's why it is used.

Reply:
I think it is fixed everywhere now.